types/msg.go

```diff
@@ -103,10 +103,47 @@ type CosmosMsg struct {
 	Gov          *GovMsg          `json:"gov,omitempty"`
 	IBC          *IBCMsg          `json:"ibc,omitempty"`
 	Staking      *StakingMsg      `json:"staking,omitempty"`
-	Stargate     *StargateMsg     `json:"stargate,omitempty"`
+	Any          *AnyMsg          `json:"any,omitempty"`
 	Wasm         *WasmMsg         `json:"wasm,omitempty"`
 }
 
+func (m *CosmosMsg) UnmarshalJSON(data []byte) error {
+	// We need a custom unmarshaler to parse both the "stargate" and "any" variants
+	type InternalCosmosMsg struct {
+		Bank         *BankMsg         `json:"bank,omitempty"`
+		Custom       json.RawMessage  `json:"custom,omitempty"`
+		Distribution *DistributionMsg `json:"distribution,omitempty"`
+		Gov          *GovMsg          `json:"gov,omitempty"`
+		IBC          *IBCMsg          `json:"ibc,omitempty"`
+		Staking      *StakingMsg      `json:"staking,omitempty"`
+		Any          *AnyMsg          `json:"any,omitempty"`
+		Wasm         *WasmMsg         `json:"wasm,omitempty"`
+		Stargate     *AnyMsg          `json:"stargate,omitempty"`
+	}
+	var tmp InternalCosmosMsg
+	err := json.Unmarshal(data, &tmp)
+	if err != nil {
+		return err
+	}
+
+	// Use "Any" for both variants
+	if tmp.Any == nil && tmp.Stargate != nil {
+		tmp.Any = tmp.Stargate
+	}
```

Comment:
I think it would be fair to return an error in case both fields are set instead of the implicit override behaviour here. Would be slightly safer and easier to review.

Reply:
Good point, better to error early in this case than to produce potentially confusing results.